src/graph.rs

```diff
@@ -905,10 +903,11 @@ impl PyGraph {
     pub fn add_edges_from_no_data(
         &mut self,
         py: Python,
-        obj_list: Vec<(usize, usize)>,
+        obj_list: Bound<'_, PyAny>,
     ) -> PyResult<EdgeIndices> {
-        let mut out_list: Vec<usize> = Vec::with_capacity(obj_list.len());
-        for obj in obj_list {
+        let mut out_list: Vec<usize> = Vec::new();
+        for py_obj in obj_list.iter()? {
```

Comment:
The main other alternatives I want to test are:
* Calling https://docs.rs/pyo3/latest/pyo3/struct.Python.html#method.is_instance first and downcasting to list as an special case
* Try extracting to a `Vec` first and then handling the case where it fails

Reply:
I'm not sure what you mean with the first alternative. The second one seems unnecessary given the current performance results. I suspect that `PyO3` is basically doing all of this under the hood when you tell it to convert a list into a `Vec<(usize, usize>)`.

Reply:
Hmmm the `is_instance` method is just the equivalent of `isintanceof(x, list)` in Python. For some reason the documentation link I copied was wrong